ui/vic-ui-h5c/vic/src/vic-app/src/app/summary-view/summary-view.component.spec.ts

```diff
@@ -0,0 +1,95 @@
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import { async, TestBed, ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { HttpModule } from '@angular/http';
+import { Observable, Subject } from 'rxjs/Rx';
+
+import {
+    AppAlertComponent,
+    AppAlertService,
+    Globals,
+    GlobalsService,
+    I18nService,
+    RefreshService
+} from '../shared/index';
+import { DataPropertyService } from '../services/data-property.service';
+import { AppErrorHandler } from '../shared/appErrorHandler';
+import { VicSummaryViewComponent } from './summary-view.component';
+
+import { ClarityModule } from 'clarity-angular';
+
+describe('VIC object view: Summary tab', () => {
+    let fixture: ComponentFixture<VicSummaryViewComponent>;
+    let compInstance: VicSummaryViewComponent;
+    let dpService: DataPropertyService;
+    let globalService: GlobalsService;
+    let refreshService: RefreshService;
+
+    class DpServiceStub {
+        private vicObjectSource = new Subject<any>();
+        public vicObject$ = this.vicObjectSource.asObservable();
+        fetchRootInfo(props: string[]): void {
+            this.vicObjectSource.next({
+                uiVersion: '1.1',
+                vchVmsLen: 1,
+                containerVmsLen: 1
+            });
+        }
+    }
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                { provide: DataPropertyService, useClass: DpServiceStub },
+                AppAlertService,
+                Globals,
+                GlobalsService,
+                RefreshService
+            ],
+            imports: [
+                HttpModule
+            ],
+            declarations: [
+                VicSummaryViewComponent
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(VicSummaryViewComponent);
+        compInstance = fixture.componentInstance;
+        dpService = fixture.debugElement.injector.get(DataPropertyService);
+        globalService = fixture.debugElement.injector.get(GlobalsService);
+        refreshService = fixture.debugElement.injector.get(RefreshService);
+    });
+
+    it('should display version and vch length correctly', () => {
+        fixture.detectChanges();
+        let ul = fixture.debugElement.query(By.css('ul.summary-items-list'));
+        let span_uiVersion = ul
+            .query(By.css('li:nth-child(2)'))
```

Comment:
This test seems very rigid. If you change the html it would fail. Can you do the lookup by class instead?